lib/features/cardio_log/presentation/screens/cardio_input_page.dart

```diff
@@ -315,60 +319,57 @@ class CardioInputPageState extends State<CardioInputPage> {
         return 'Swim';
     }
   }
-  
+
   // Function to save activity
   Future<void> _saveActivity(double calories) async {
     try {
       CardioActivity? activity;
-      DateTime now = DateTime.now();
-      
+
       switch (selectedType) {
         case CardioType.running:
+          // Access form state directly using the key
+          final formState = _runningFormKey.currentState!;
+
```

Comment:
```suggestion
          final formState = _runningFormKey.currentState!;
// This can be improved with null safety check, if the state could be null
final formState = _runningFormKey.currentState;
if (formState != null) {
  // Proceed with formState
}
```

Akan lebih baik ditambahkan null safety check untuk apabila state bernilai null